guides/features/pipecat-flows.mdx

```diff
@@ -466,517 +292,273 @@ NodeConfig(
 
-### Best Practices
+### Function Calls
 
-- Store shared data in flow_manager.state
-- Create separate functions for node creation
+Pipecat Flows supports three ways to define function calls:
 
-# Flow Editor
+#### FlowsFunctionSchema (Recommended)
 
-The Pipecat Flow Editor provides a visual interface for creating and managing conversation flows. It offers a node-based interface that makes it easier to design, visualize, and modify your flows.
+Provider agnostic function call definition that works across all LLM providers:
 
-<Frame>![Food Ordering Flow](/images/food-ordering-flow.png)</Frame>
+```python
+from pipecat_flows import FlowsFunctionSchema
 
-## Visual Design
+collect_info_func = FlowsFunctionSchema(
+    name="collect_user_info",
+    description="Collect basic user information",
+    required=["name", "email"],
+    handler=process_user_info,
+    properties={
+        "name": {"type": "string"},
+        "email": {"type": "string", "format": "email"}
+    },
+)
+```
 
-### Node Types
+#### FlowsDirectFunction
 
-- **Start Node** (Green): Entry point of your flow
+Consolidates the function definition and handler logic into a single function. The function signature and docstring are automatically used to generate the function schema:
 
-  ```python
-  "greeting": {
-      "role_messages": [...],
-      "task_messages": [...],
-      "functions": [...]
-  }
-  ```
+```python
+async def collect_user_info(
+    flow_manager: FlowManager,
+    name: str,
+    email: str
+) -> tuple[FlowResult, NodeConfig]:
+    """
+    Collect basic user information.
 
-- **Flow Nodes** (Blue): Intermediate states
+    Args:
+        name (str): The user's full name
+        email (str): The user's email address
+    """
+    # Process the information
+    flow_manager.state["user_info"] = {"name": name, "email": email}
 
-  ```python
-  "collect_info": {
-      "task_messages": [...],
-      "functions": [...],
-      "pre_actions": [...]
-  }
-  ```
+    # Return result and next node
+    return {"status": "success"}, create_next_node()
 
-- **End Node** (Red): Final state
+# Use directly in NodeConfig
+node_config = {
+    "functions": [collect_user_info]
+}
+```
 
-  ```python
-  "end": {
-      "task_messages": [...],
-      "functions": [],
-      "post_actions": [{"type": "end_conversation"}]
-  }
-  ```
+#### Provider Native Format
 
-- **Function Nodes**:
-  - Edge Functions (Purple): Create transitions
-    ```python
-    {
-        "name": "next_node",
-        "description": "Transition to next state"
-    }
-    ```
-  - Node Functions (Orange): Perform operations
-    ```python
-    {
-        "name": "process_data",
-        "handler": process_data_handler,
-        "description": "Process user data"
-    }
-    ```
+Uses the provider's native format (OpenAI, Anthropic, etc.). While supported, we recommend using FlowsFunctionSchema for better portability.
```

Comment:
👍 I like how here (and above, with the static flows example), we're cutting down on the time we spend on non-recommended ways of doing things.